app/routers/meds.py

```diff
@@ -0,0 +1,424 @@
+    """
+    early_datetime = datetime.combine(day, early)
+    late_datetime = datetime.combine(day, late)
+    late_datetime = adjust_day(late_datetime, early, late)
+    return early_datetime <= reminder <= late_datetime
+
+
+def validate_first_day_reminder(previous: datetime, reminder_time: time,
+                                minimum: time, maximum: time) -> bool:
+    """Returns True if interval between reminders is valid, false otherwise.
+
+    Args:
+        previous (datetime): Previous reminder.
+        reminder_time (time): New reminder time.
+        minimum (time) - Minimal interval between reminders.
+        maximum (time) - Maximal interval between reminders.
+
+    Returns:
+        bool: True if interval between reminders is valid, false otherwise.
+    """
+    interval = get_interval_in_minutes(previous.time(), reminder_time)
+    min_minutes = convert_time_to_minutes(minimum)
+    max_minutes = convert_time_to_minutes(maximum)
+    return max_minutes >= interval >= min_minutes
+
+
+def get_different_time_reminder(previous: datetime, minimum: time, early: time,
+                                late: time) -> Optional[datetime]:
+    """Returns datetime object for first day reminder with non-standard time.
+
+    Args:
+        previous (datetime): Previous reminder.
+        minimum (time) - Minimal interval between reminders.
+        early (time): Earliest reminder time.
+        late (late): Latest reminder time. Interpreted as following day if
+                     earlier than early.
+
+    Returns:
+        datetime | None: First day reminder with non-standard time, if valid.
+    """
+    reminder = previous + timedelta(minutes=convert_time_to_minutes(minimum))
+    if validate_datetime(reminder, previous.date(), early, late):
+        return reminder
+
+
+def get_first_day_reminders(form: Dict[str, str],
+                            times: List[time]) -> Iterator[datetime]:
+    """Generates datetime objects for reminders on the first day.
+
+    Args:
+        form (dict(str, str)): Medication form containing all relevant data.
+        times (list(time)): Time objects for reminders.
+
+    Yields:
+        datetime: First day reminder datetime object.
+    """
+    _, _, amount, early, late, minimum, maximum, _, start, _ = trans_form(form)
+    yield start
+    datetime_obj = start
+    i = 1
+    for time_obj in times:
+        if i <= amount:
+            reminder = datetime.combine(start.date(), time_obj)
+            reminder = adjust_day(reminder, early, time_obj)
+            if reminder > start:
+                if not validate_first_day_reminder(datetime_obj, time_obj,
+                                                   minimum, maximum):
+                    reminder = get_different_time_reminder(
+                        datetime_obj, minimum, early, late)
+                yield reminder
+                datetime_obj = reminder
+                i += 1
```

Comment:
This block should be in a separate function.
It would allow you to put guards and create less indented, more fluent and readable code :)